src/_blog/libp2p-hole-punching.md

```diff
@@ -37,130 +36,128 @@ The simplest security measure all major consumer firewalls enforce to protect th
     2.  If there is not, drop the packet.
 
 ## An example
 
-Let's look at an example. Two computers, *A* and *B*, each in their own network, connected to the internet via a separate home router each, with those home routers acting as firewalls.
+Let's look at an example. Two computers, *A* and *B*, each in their own network, are connected to the internet via a separate home router each, with those home routers acting as firewalls.
 
 ![img](../assets/libp2p-hole-punching-network.svg)
 
-The sequence diagram below depicts the scenario where computer *A* emitts a packet destined for *B*. Said packet is first send to *A*'s router, which in turn forwards it to *B*'s router.
+The sequence diagram below depicts the scenario where computer *A* emits a packet destined for *B*. Said packet is first sent to *A*'s router, which in turn forwards it to *B*'s router.
 
 *A*'s router forwards the packet to *B*'s router. *B*'s router checks its state table, cannot find a matching 5-tuple (IP<sub>A</sub>, IP<sub>B</sub>, TCP, Port<sub>A</sub>, Port<sub>B</sub>) and thus drops the packet. In a nutshell, this is why the Internet is divided into two, public and non-public computers.
 
-Same applies to packets send from *B* to *A*, see second half of the sequence diagram.
+The same applies to packets sent from *B* to *A*; see the second half of the sequence diagram.
 
 ![img](../assets/libp2p-hole-punching-firewall.svg)
 
-Now this should not suggest to go ahead and disable all firewalls across the world. Please no. They do serve their purpose. Afterall *A* and *B* most likely don't want random strangers connecting to them. Though they do still want to connect to each other.
+Now, please do not go around disabling all your firewalls! They serve their purpose. After all, *A* and *B* most likely don't want random strangers connecting to them. However, they do still want to connect to each other.
 
 # Hole Punching
 
 Hole punching is one mechanism to overcome the problem described above. Let's see how it works.
 
-Imagine that we have some mysterious mechanism to synchronize *A* and *B*. Mysterious as synchronizing clocks is hard, &#x2026; really hard. For those that don't believe me, I recommend reading Lamport's [_Time, Clocks, and the Ordering of Events in a Distributed System_](https://web.archive.org/web/20211101121036/https://amturing.acm.org/p558-lamport.pdf) and Mills' [_A Brief History of NTP Time: Memoirs of an Internet Timekeeper_](https://web.archive.org/web/20220120075303/https://www.eecis.udel.edu/~mills/database/papers/history.pdf).
-
-Anyways, back to assuming the existence of a mysterious synchronization mechanism. Such mechanism would allow *A* and *B* to dial each other "at the same time".
+Imagine that we have some mysterious mechanism to synchronize *A* and *B*. Such a mechanism would allow *A* and *B* to dial each other "at the same time".
```

Comment:
Why remove these further reading suggestions @johnnymatthews?

Reply:
Ah, my bad. Didn't meant to take it out. But I'm a bit confused why it's in here. What does this _mysterious clock synchronisation_ have to do with the rest of this post? I might be missing something, but this felt very out of place, as was only put in here to mention these two articles from Lamport and Mills.

Reply:
That makes sense. I think these are great papers to read on time synchronization. But as you say, time synchronization only plays a minor role in this blog post, thus referencing these is too much.

(Also don't want to come across as just _name dropping_.)

Thanks @johnnymatthews.